### ISSUE TITLE
`prefer-stateless-function` should detect `null` in if shorthand return

### ISSUE THREAD

**Issue by inyono** [yannickcr/eslint-plugin-react#516]
With enabled rule `prefer-stateless-function`, the following does throw a warning although `null` may be returned:
```.js
class If extends Component {
  render() {
    const { condition, ...props } = this.props

    return condition ? <this.props.component {...props} /> : null
  }
}
```
If we write the result to a variable, though, there is no warning:
```.js
class If extends Component {
  render() {
    const { condition, ...props } = this.props

    const component = condition ? <this.props.component {...props} /> : null

    return component
  }
}
```
